### PR TITLE
Updated alert vue component

### PR DESCRIPTION
**Pull request by 98dfsj**
# Closes issue: EN-450

## Overview

Upgraded the Alert Vue Component to work as expected.

## Work carried out

- [x] Upgraded component to behave as a modal
- [x] Added tests
- [x] Updated styles

### CODE REVIEW

File: packages/css-framework/src/components/_alert.scss [hunk 1]
```diff
@@ -46,6 +46,11 @@ $alert-states: (
     margin: 0;
   }
 
+  opacity: 0;
```
Comment (antictor) [inline packages/css-framework/src/components/_alert.scss]: Can these styles move to css framework so they are shared by vue, react and whatever the user chooses to use?

File: packages/vue-component-library/src/components/RnAlert/RnAlert.spec.js [hunk 2]
```diff
@@ -8,7 +8,11 @@ describe('RnAlert.vue', () => {
   beforeEach(() => {
     wrapper = shallow(RnAlert, {
       propsData: {
-        text: 'foo',
```
Comment (antictor) [inline packages/vue-component-library/src/components/RnAlert/RnAlert.spec.js]: Do we have tests to check the default props are correct?

File: packages/vue-component-library/src/components/RnAlert/RnAlert.spec.js [hunk 3]
```diff
@@ -24,4 +28,26 @@ describe('RnAlert.vue', () => {
...
+  it('Displays an action button', () => {
+    expect(wrapper.find('#action-button').text()).toBe('ok')
+  })
+
+  it('Displays a close button', () => {
+    expect(wrapper.find('#close-button').isVisible()).toBe(true)
+  })
+
+
 })
```
Comment (antictor) [inline packages/vue-component-library/src/components/RnAlert/RnAlert.spec.js]: Are there tests to test interaction with the buttons etc to make ensure clicking close actually closes it?

File: packages/vue-component-library/src/components/RnAlert/index.vue [hunk 4]
```diff
@@ -1,19 +1,30 @@
...
-      <svg xmlns="http://www.w3.org/2000/svg" width="12" height="12">
-        <path
-          fill="currentColor"
-          fillRule="evenod"
-          d="M8.09 6l3.48 3.48a1.48 1.48 0 1 1-2.09 2.09L6 8.09l-3.48 3.48A1.48 1.48 0 0 1 .43 9.48L3.91 6 .43 2.52A1.48 1.48 0 1 1 2.52.43L6 3.91 9.48.43a1.48 1.48 0 0 1 2.09 2.09L8.09 6z"
-        />
-      </svg>
-    </button>
+  <div class="rn-alert" :class="[state, error]">
+    <div class="window">
```
Comment (antictor) [inline packages/vue-component-library/src/components/RnAlert/index.vue]: could .window be confused with the global browser window?

File: packages/css-framework/src/components/_alert.scss (hunk 1 above)
Comment (98dfsj) [reply packages/css-framework/src/components/_alert.scss]: They are in the CSS Framework?

File: packages/vue-component-library/src/components/RnAlert/RnAlert.spec.js (hunk 2 above)
Comment (98dfsj) [reply packages/vue-component-library/src/components/RnAlert/RnAlert.spec.js]: Do you mean like this?

```
  it('Has expected props', () => {
    expect(wrapper.props('title')).toBe('This is a test title')
    expect(wrapper.props('actionButtonText')).toBe('ok')
    expect(wrapper.props('hideClose')).toBe(false)
    expect(wrapper.props('hideAction')).toBe(false)
    expect(wrapper.props('error')).toBe(false)
  })
```

If so, that's already in there. If not, then what do you mean?

File: packages/vue-component-library/src/components/RnAlert/RnAlert.spec.js (hunk 3 above)
Comment (98dfsj) [reply packages/vue-component-library/src/components/RnAlert/RnAlert.spec.js]: No, that's a fair point. I can add those.

File: packages/vue-component-library/src/components/RnAlert/index.vue (hunk 4 above)
Comment (98dfsj) [reply packages/vue-component-library/src/components/RnAlert/index.vue]: Seems unlikely. Especially with it being a css class.

File: packages/css-framework/src/components/_alert.scss (hunk 1 above)
Comment (antictor) [reply packages/css-framework/src/components/_alert.scss]: yeah I see that ;p

File: packages/vue-component-library/src/components/RnAlert/RnAlert.spec.js (hunk 2 above)
Comment (antictor) [reply packages/vue-component-library/src/components/RnAlert/RnAlert.spec.js]: The test here provides the prop values, you should also have a test that does not send all the props to check the the defaults are correctly set

File: packages/vue-component-library/src/components/RnAlert/index.vue (hunk 4 above)
Comment (antictor) [reply packages/vue-component-library/src/components/RnAlert/index.vue]: I know.. just wondering if there is a better name to use

Comment (98dfsj) [reply packages/vue-component-library/src/components/RnAlert/index.vue]: I've renamed it to `alert_window`

**Comment by 98dfsj** [comment]
Updated styles and replaced the screenshot in this PR to reflect that.

**Comment by 98dfsj** [comment]
A gif of the component? There is one in the PR already?

**Comment by antictor** [comment]
I'm an idiot

**Pull request merged**